compiler/noirc_evaluator/src/ssa/ir/instruction/binary.rs

```diff
@@ -501,14 +502,44 @@ fn convert_signed_integer_to_field_element(int: i128, bit_size: u32) -> FieldEle
         FieldElement::from(int)
     } else {
         // We add an offset of `bit_size` bits to shift the negative values into the range [2^(bitsize-1), 2^bitsize)
+        assert!(bit_size < 128);
         let offset_int = (1i128 << bit_size) + int;
         FieldElement::from(offset_int)
     }
 }
 
 fn truncate(int: u128, bit_size: u32) -> u128 {
```

Comment:
`truncate` could use some docs, like it's not exactly clear if it clears up to or over `bit_size`.